### PR TITLE
docs(python): Adding examples to binary functions

### PR DESCRIPTION
**Pull request by datapythonista**
Adding examples to some of the binary functions. I think it should make easy to understand how to deal with binary data, even if most of the functions themselves are somehow trivial (`starts_with`...). The example is the same in the 3 contains/starts_with/ends_with functions.

### CODE REVIEW

**Comment by datapythonista** [comment]
Not really sure what's going on. Got the same exact version of black locally and in the CI, and locally doesn't want to reformat anything. Also, changes are just docs, that I don't think black should want to reformat. I'll have a look at it later.

File: py-polars/polars/expr/binary.py [hunk 1]
```diff
@@ -30,30 +30,128 @@ def contains(self, literal: bytes) -> Expr:
...
+        --------
+        starts_with : Check if the binary substring exists at the start
+        contains : Check if the binary substring exists anywhere
+
+        Examples
+        --------
+        >>> colors = pl.DataFrame(
+        ...     {
+        ...         'name': ['black', 'yellow', 'blue'],
+        ...         'code': [b'\x00\x00\x00', b'\xff\xff\x00', b'\x00\x00\xff']
```
Comment (stinodego) [inline py-polars/polars/expr/binary.py:89]: Looks like this is missing a trailing comma. Try running `blackdoc`, that should fix it.

File: py-polars/polars/expr/binary.py [hunk 2]
```diff
@@ -92,6 +190,29 @@ def encode(self, encoding: TransferEncoding) -> Expr:
...
+        ...     {
+        ...         "name": ["black", "yellow", "blue"],
+        ...         "code": [b"\x00\x00\x00", b"\xff\xff\x00", b"\x00\x00\xff"],
+        ...     }
+        ... )
+        >>> colors.select(
+        ...     "name",
+        ...     "code",
+        ...     pl.col("code").bin.encode("hex").alias("code_encoded_hex"),
+        ... )
```
Comment (stinodego) [inline py-polars/polars/expr/binary.py:205]: It's cleaner to use `with_columns` here.

**Review by stinodego — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just one minor comment then we can merge this.

File: py-polars/polars/expr/binary.py (hunk 2 above)
Comment (datapythonista) [reply py-polars/polars/expr/binary.py]: Good point, I didn't realize. Updated, thanks for the feedback and the review!

**Review by stinodego — APPROVED** [review]
Great, thanks!

**Pull request merged**